### ISSUE TITLE
fluent-bit 1.8.3 occasional crashes (caught signal SIGSEGV)

### ISSUE THREAD

**Issue by psyhomb** [fluent/fluent-bit#3955]
## Bug Report

**Describe the bug**

Fluent Bit running as systemd unit occasional crashes without being able to automatically recover, the only way to recover it is by removing DB and WAL files and then manually restarting the service unit.

```plain
Aug 15 07:44:53 test-instance-1 td-agent-bit[16104]: [2021/08/15 07:44:53] [error] [multiline] invalid stream_id 1095162821569803272, could not append content to multiline context
Aug 15 07:44:54 test-instance-1 td-agent-bit[16104]: [2021/08/15 07:44:54] [error] [multiline] invalid stream_id 1095162821569803272, could not append content to multiline context
Aug 15 07:44:54 test-instance-1 td-agent-bit[16104]: [2021/08/15 07:44:54] [error] [multiline] invalid stream_id 14222019020935586222, could not append content to multiline context
Aug 15 07:44:55 test-instance-1 td-agent-bit[16104]: [2021/08/15 07:44:55] [error] [multiline] invalid stream_id 1095162821569803272, could not append content to multiline context
Aug 15 07:44:58 test-instance-1 td-agent-bit[16104]: [2021/08/15 07:44:58] [error] [multiline] invalid stream_id 14222019020935586222, could not append content to multiline context
Aug 15 07:44:59 test-instance-1 td-agent-bit[16104]: [2021/08/15 07:44:59] [engine] caught signal (SIGSEGV)
Aug 15 07:44:59 test-instance-1 td-agent-bit[16104]: ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)
Aug 15 07:44:59 test-instance-1 systemd[1]: td-agent-bit.service: main process exited, code=killed, status=6/ABRT
Aug 15 07:44:59 test-instance-1 systemd[1]: Unit td-agent-bit.service entered failed state.
Aug 15 07:44:59 test-instance-1 systemd[1]: td-agent-bit.service failed.
Aug 15 07:44:59 test-instance-1 systemd[1]: td-agent-bit.service holdoff time over, scheduling restart.
Aug 15 07:44:59 test-instance-1 systemd[1]: Stopped TD Agent Bit.
Aug 15 07:44:59 test-instance-1 systemd[1]: Starting TD Agent Bit...
Aug 15 07:44:59 test-instance-1 systemd[1]: Started TD Agent Bit.
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: Fluent Bit v1.8.3
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: * Copyright (C) 2019-2021 The Fluent Bit Authors
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: * Copyright (C) 2015-2018 Treasure Data
--
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: [2021/08/15 07:44:59] [ info] [input:tail:tail.27] multiline core started
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: [2021/08/15 07:44:59] [ info] [input:tail:tail.28] multiline core started
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: [2021/08/15 07:44:59] [ info] [input:tail:tail.29] multiline core started
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: [2021/08/15 07:44:59] [ info] [input:tail:tail.30] multiline core started
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: [2021/08/15 07:44:59] [ info] [input:tail:tail.31] multiline core started
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: [2021/08/15 07:44:59] [ info] [input:tail:tail.32] multiline core started
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: [2021/08/15 07:44:59] [ info] [sp] stream processor started
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: [2021/08/15 07:44:59] [ info] [input:tail:tail.3] inotify_fs_add(): inode=1185289 watch_fd=1 name=/var/log/test/service-1/request.log
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: [2021/08/15 07:44:59] [ info] [input:tail:tail.3] inotify_fs_add(): inode=1185429 watch_fd=2 name=/var/log/test/service-1/service.log
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: [2021/08/15 07:44:59] [engine] caught signal (SIGSEGV)
Aug 15 07:44:59 test-instance-1 td-agent-bit[20430]: ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)
Aug 15 07:44:59 test-instance-1 systemd[1]: td-agent-bit.service: main process exited, code=killed, status=6/ABRT
Aug 15 07:44:59 test-instance-1 systemd[1]: Unit td-agent-bit.service entered failed state.
Aug 15 07:44:59 test-instance-1 systemd[1]: td-agent-bit.service failed.
Aug 15 07:44:59 test-instance-1 systemd[1]: td-agent-bit.service holdoff time over, scheduling restart.
Aug 15 07:44:59 test-instance-1 systemd[1]: Stopped TD Agent Bit.
Aug 15 07:44:59 test-instance-1 systemd[1]: Starting TD Agent Bit...
Aug 15 07:44:59 test-instance-1 systemd[1]: Started TD Agent Bit.
Aug 15 07:44:59 test-instance-1 td-agent-bit[20489]: Fluent Bit v1.8.3
Aug 15 07:44:59 test-instance-1 td-agent-bit[20489]: * Copyright (C) 2019-2021 The Fluent Bit Authors
Aug 15 07:44:59 test-instance-1 td-agent-bit[20489]: * Copyright (C) 2015-2018 Treasure Data
--
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: [2021/08/15 07:44:59] [ info] [input:tail:tail.28] multiline core started
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: [2021/08/15 07:44:59] [ info] [input:tail:tail.29] multiline core started
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: [2021/08/15 07:44:59] [ info] [input:tail:tail.30] multiline core started
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: [2021/08/15 07:44:59] [ info] [input:tail:tail.31] multiline core started
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: [2021/08/15 07:44:59] [ info] [input:tail:tail.32] multiline core started
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: [2021/08/15 07:45:00] [ info] [sp] stream processor started
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: [2021/08/15 07:45:00] [ info] [input:tail:tail.3] inotify_fs_add(): inode=1185289 watch_fd=1 name=/var/log/test/service-1/request.log
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: [2021/08/15 07:45:00] [ info] [input:tail:tail.3] inotify_fs_add(): inode=1185429 watch_fd=2 name=/var/log/test/service-1/service.log
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: [2021/08/15 07:45:00] [ info] [input:tail:tail.4] inotify_fs_add(): inode=1185422 watch_fd=1 name=/var/log/test-2/service-1/request.log
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: [2021/08/15 07:45:00] [engine] caught signal (SIGSEGV)
Aug 15 07:45:00 test-instance-1 td-agent-bit[20489]: ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)
```

**Steps to reproduce the problem**

There is no easy way to reproduce it, because it is not quite clear to me what is causing the issue.

**Expected behavior**

Be able to automatically recover by systemd (`Restart=always`) without removing DB and WAL files.

**Your Environment**
* Version used: `1.8.3`
* Environment name and version (e.g. Kubernetes? What version?): `systemd 219`
* Server type and version: `AWS EC2`
* Operating System and version: `Amazon Linux 2`
* Filters and plugins: `record_modifier` and `none`
* Configuration:
```ini
[INPUT]
    name                tail
    path                /var/log/test/service-1/*.log
    tag                 test-service-1
    multiline.parser    multiline_java

    Skip_Long_Lines     on
    Refresh_Interval    5
    Rotate_Wait         10

    Buffer_Chunk_Size   128KB
    Buffer_Max_Size     5MB
    Mem_Buf_Limit       100MB

    DB                  /tmp/test-td-agent-bit.db
    DB.sync             normal
    DB.locking          true
    DB.journal_mode     WAL


########################################################################################################


[FILTER]
    Name                record_modifier
    Match               test-service-1
    Record              source file
    Record              hostname ${HOSTNAME}
    Record              ec2_instance_id ${EC2_INSTANCE_ID}
    Record              service_name test-service-1


########################################################################################################


[OUTPUT]
    name                    http
    match                   *-service-1
    host                    c.example.com
    port                    443
    tls                     on

    http_user               ${CUSTOMER_NAME}
    http_passwd             ${CUSTOMER_TOKEN}
    uri                     /v1/http/fluentbit

    format                  msgpack
    header                  X-Example-Decoder fluent
    compress                gzip
    log_response_payload    true
    Retry_Limit             2
```

**Additional context**

We have to automate restarts (remove DB and WAL files and then restart the service) to prevent further dropping of logs data.

**Comment by psyhomb** [comment]
Had all kind of different issues since day one with fluent-bit and I have it enough, I've simply switched to  Filebeat OSS and everything is running perfectly smooth.

**Comment by psyhomb** [comment]
Anything new on this, is this issue resolved?

**Comment by nokute78** [comment]
I sent a patch https://github.com/fluent/fluent-bit/pull/4197 to fix `invalid stream_id` error.
The patch is merged from v1.8.9. 
https://fluentbit.io/announcements/v1.8.9/

Could you check it ?
